### PR TITLE
Enhancement: Raise error level from 7 to 6

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] raises the error level for `vimeo/psalm` from `7` to `6`
* [x] fixes issues reported via static code analysis
* [x] regenerates the baseline 

Follows #1897.

### CODE REVIEW

File: src/Faker/Provider/ar_SA/Address.php [hunk 1]
```diff
@@ -96,14 +96,6 @@ class Address extends \Faker\Provider\Address
 
     protected static $secondaryAddressFormats = ['شقة رقم. ##', 'عمارة رقم ##'];
 
-    /**
-     * @example 'شرق'
-     */
-    public static function cityPrefix()
-    {
-        return static::randomElement(static::$cityPrefix);
```
Comment (localheinz) [inline src/Faker/Provider/ar_SA/Address.php:104]: This property does not exist.

File: src/Faker/Provider/ja_JP/Text.php [hunk 2]
```diff
@@ -622,7 +622,7 @@ protected static function appendEnd($text)
         if (function_exists('mb_substr')) {
             $last = mb_substr($text, 0, mb_strlen($text) - 1, 'UTF-8');
         } else {
-            $chars = static::split($text);
```
Comment (localheinz) [inline src/Faker/Provider/ja_JP/Text.php:625]: This method does not exist.

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
tests fail

**Comment by localheinz** [comment]
@fzaninotto 

Adjusted!

**Comment by localheinz** [comment]
Thank you, @fzaninotto!

**Pull request merged**